frontend/components/common/InstantSearchProvider.tsx

```diff
@@ -169,7 +169,16 @@ export function InstantSearchProvider({
                }
             );
 
-            return `${baseUrl}${path}${queryString}`;
+            let customQueries = '';
+            const customQueryParams = new URLSearchParams(location.search);
+            if (customQueryParams.has('_vercel_no_cache')) {
+               customQueries =
+                  (queryString ? '&' : '?') +
+                  '_vercel_no_cache=' +
+                  customQueryParams.get('_vercel_no_cache');
+            }
+
+            return `${baseUrl}${path}${queryString}${customQueries}`;
```

Comment:
What you have seems like it works. However, I'd rather pass everything through `qsModule` just to be safe, so we don't get mixed up with `&` and `?` and url encoding and such.

```ts
            const params = new URLSearchParams(location.search);
            const vercelCache = params.has('_vercel_no_cache')
               ? { _vercel_no_cache: params.get('_vercel_no_cache') }
               : {};

            const queryString = qsModule.stringify(
               { ...routeState, filter: filterCopy, ...vercelCache },
               {
                  addQueryPrefix: true,
                  arrayFormat: 'indices',
               }
            );

            return `${baseUrl}${path}${queryString}`;
```

Reply:
I prefer @andyg0808's solution actually.